sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/Scan.java

```diff
@@ -115,4 +116,14 @@ default ContinuousStream toContinuousStream(String checkpointLocation) {
   default CustomMetric[] supportedCustomMetrics() {
     return new CustomMetric[]{};
   }
+
+  /**
+   * Return an array of custom metrics which are collected with values at the driver side only.
+   * Note that these metrics must be included in the supported custom metrics reported by
+   * `supportedCustomMetrics`.
+   */
+  default CustomTaskMetric[] reportCustomDriverMetrics() {
```

Comment:
maybe `reportDriverMetrics`? there is also no `custom` in `PartitionReader.currentMetricsValues`

Reply:
sounds good.

Reply:
renamed.